content/docs/courses/fundamentals/zero-single-sign-on.mdx

```diff
@@ -0,0 +1,246 @@
+
+With Pomerium, the ability to mint, sign, and forward a JWT as an HTTP request header is native to the product – and, it's as simple as making one configuration change in the Zero Console.
+
+Now, let's see how SSO with JWTs works with Pomerium:
+
+![Diagram depicting the JWT flow for single sign on with Pomerium](./img/zero-sso/jwt-flow.svg)
+
+1. End-user accesses the upstream service
+
+1. Pomerium authenticates the end-user against an identity provider and creates a Pomerium JWT
+
+1. Pomerium authorizes the user, then signs and forwards the JWT Assertion Header to the upstream service
+
+1. Upstream service fetches the public key from the JWKS endpoint, validates the signature, and grants the end-user access
+
+In this way, the upstream application is secured on the protocol layer with an encrypted TLS connection, and on the application layer with JWT authentication. The user is also subject to the rules of the authorization policy.
+
+Now, let's configure Pomerium to forward the JWT.
+
+## Configure SSO
+
+### Apply Pass Identity Headers
+
+In the Zero Console:
+
+1. Select **Routes**
+1. Select your Grafana route
+1. Select the **Headers** tab
+1. Apply the **Pass Identity Headers** setting
+
+   ![Apply the route-level pass identity headers setting in the Zero Console](./img/zero-sso/pz-courses-apply-pass-id-headers.gif)
+
+:::info
+
+The [**Pass Identity Headers**](/docs/reference/routes/pass-identity-headers-per-route) setting instructs Pomerium to forward the signed `x-pomerium-jwt-assertion` HTTP header to the upstream application.
+
+:::
+
+### Add Grafana JWT settings
+
+In your Docker Compose file, add the following environment variables to your Grafana service:
+
+```yaml title="docker-compose.yaml" {8-15} showLineNumbers
+grafana:
+  image: grafana/grafana:latest
+  ports:
+    - 3000:3000
+  networks:
+    main: {}
+  environment:
+    - GF_AUTH_SIGNOUT_REDIRECT_URL=https://grafana.<CLUSTER_SUBDOMAIN>.pomerium.app/.pomerium/sign_out
+    - GF_AUTH_JWT_ENABLED=true
+    - GF_AUTH_JWT_HEADER_NAME=X-Pomerium-Jwt-Assertion
+    - GF_AUTH_JWT_EMAIL_CLAIM=email
+    - GF_AUTH_JWT_USERNAME_CLAIM=sub
+    - GF_AUTH_JWT_JWK_SET_URL=https://authenticate.<CLUSTER_SUBDOMAIN>.pomerium.app/.well-known/pomerium/jwks.json
+    - GF_AUTH_JWT_CACHE_TTL=60m
+    - GF_AUTH_JWT_AUTO_SIGN_UP=true
+  volumes:
+    - ./grafana-storage:/var/lib/grafana
+```
+
+> See Grafana's [Configure JWT authentication](https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/jwt/) docs for more information about these settings.
+
+:::note
+
+In lines `8` and `13`, replace `CLUSTER_SUBDOMAIN` with your cluster subdomain.
+
+For example, if your starter domain is `curious-cat-9999.pomerium.app`, your cluster subdomain would be `curious-cat-9999`.
+
+:::
+
+You also need to add another alias for the Pomerium Authenticate service:
+
+```yaml title="docker-compose"
+  pomerium:
+    image: pomerium/pomerium:v0.25.1
```

Comment:
Why this we use **pomerium/pomerium:v0.26.0,** 
but in other places **cr.pomerium.com/pomerium/pomerium:v0.26.0**

Reply:
The `cr.pomerium.com/pomerium/pomerium:vX.XX.X` URL is a tracking URL. We don't use it in the onboarding scripts.